flaky test: TestAPI_AgentMonitorJSON

https://app.circleci.com/pipelines/github/hashicorp/consul/8152/workflows/c70432ec-d0a3-467e-9729-737575bcba60/jobs/148034

These logs aren't super helpful since there is output from other tests along with the failed one: 
```
Failed
=== RUN   TestAPI_AgentMonitorJSON
=== PAUSE TestAPI_AgentMonitorJSON
=== CONT  TestAPI_AgentMonitorJSON
    TestAgentService_Register_TerminatingGateway: io.go:407: 2020-03-27T19:09:55.986Z [INFO]  agent: Synced node info
    TestAPI_CatalogNodes_MetaFilter: io.go:407: 2020-03-27T19:09:55.986Z [WARN]  agent.server.serf.wan: serf: Shutdown without a Leave
    TestAPI_CatalogServiceCached: io.go:407: 2020-03-27T19:09:55.986Z [DEBUG] agent.http: Request finished: method=GET url=/v1/catalog/service/consul?cached= from=127.0.0.1:56458 latency=167.532µs
    TestAPI_CatalogNodes: io.go:407: 2020-03-27T19:09:55.986Z [DEBUG] agent.http: Request finished: method=GET url=/v1/health/node/node-c7bd0826-916d-e663-3b47-8443af9c6e79 from=127.0.0.1:59674 latency=274.949µs
    TestAPI_AgentMonitorJSON: server.go:278: CONFIG JSON: {"node_name":"node-d86bac4f-7f85-b838-8d13-931f7c2d7b7c","node_id":"d86bac4f-7f85-b838-8d13-931f7c2d7b7c","performance":{"raft_multiplier":1},"bootstrap":true,"server":true,"data_dir":"/tmp/TestAPI_AgentMonitorJSON184590684/data","segments":null,"disable_update_check":true,"log_level":"debug","bind_addr":"127.0.0.1","addresses":{},"ports":{"dns":14759,"http":14760,"https":14761,"serf_lan":14762,"serf_wan":14763,"server":14764},"acl_enforce_version_8":false,"acl":{"tokens":{}},"connect":{"ca_config":{"cluster_id":"11111111-2222-3333-4444-555555555555"},"enabled":true}}
--- FAIL: TestAPI_AgentMonitorJSON (13.02s)
```

--- a/api/agent_test.go
+++ b/api/agent_test.go
@@ -1209,16 +1209,27 @@ func TestAPI_AgentMonitorJSON(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
-	// Wait for the first log message and validate it is valid JSON
-	select {
-	case log := <-logCh:
-		var output map[string]interface{}
-		if err := json.Unmarshal([]byte(log), &output); err != nil {
-			t.Fatalf("log output was not JSON: %q", log)
+	retry.Run(t, func(r *retry.R) {
+		{
+			// Register a service to be sure something happens in secs
+			serviceReg := &AgentServiceRegistration{
+				Name: "redis",
+			}
+			if err := agent.ServiceRegister(serviceReg); err != nil {
+				r.Fatalf("err: %v", err)
+			}
 		}
-	case <-time.After(10 * time.Second):
-		t.Fatalf("failed to get a log message")
-	}
+		// Wait for the first log message and validate it is valid JSON
+		select {
+		case log := <-logCh:
+			var output map[string]interface{}
+			if err := json.Unmarshal([]byte(log), &output); err != nil {
+				r.Fatalf("log output was not JSON: %q", log)
+			}
+		case <-time.After(10 * time.Second):
+			r.Fatalf("failed to get a log message")
+		}
+	})
 }
 
 func TestAPI_ServiceMaintenance(t *testing.T) {

